### PR TITLE
Code optimization

### PR DESCRIPTION
**Pull request by HereAround**
Contains the changes in https://github.com/Julia-meets-String-Theory/FTheoryTools.jl/pull/10.

### CODE REVIEW

File: docs/src/FTheoryTools/Tate.md [hunk 1]
```diff
@@ -24,27 +26,79 @@ Then form a ``\mathbb{P}^{2,3,1}``-bundle over ``B3`` such that
...
+tate_section_a1(t::GlobalTateModel)
+tate_section_a2(t::GlobalTateModel)
+tate_section_a3(t::GlobalTateModel)
+tate_section_a4(t::GlobalTateModel)
+tate_section_a6(t::GlobalTateModel)
+tate_polynomial(t::GlobalTateModel)
+cy_hypersurface(t::GlobalTateModel)
+```
+In case the Tate model is constructed over a not fully specified base, it is
+nontheless possibly to construct an auxiliary base space as well as an
```
Comment (apturner) [inline docs/src/FTheoryTools/Tate.md:88]:
```suggestion
nonetheless possible to construct an auxiliary base space as well as an
```

File: docs/src/FTheoryTools/Weierstrass.md [hunk 2]
```diff
@@ -20,23 +22,81 @@ Then form a ``\mathbb{P}^{2,3,1}``-bundle over ``B3`` such that
 - ``z`` transforms as a section of ``0 \overline{K}_{B_3} = \mathcal{O}_{B_3}``.
 In this 5-fold ambient space, a global Weierstrass model is the hypersurface defined
 by the vanishing of the Weierstrass polynomial ``P_W = x^3 - y^2 + f x z^4 + g z^6``.
-For such geometries, we support functionality.
 
-## Constructors
+To construct a Weierstrass model as hypersurface in an ambient space,
```
Comment (apturner) [inline docs/src/FTheoryTools/Weierstrass.md:26]:
```suggestion
To construct a Weierstrass model as a hypersurface in an ambient space,
```

File: docs/src/FTheoryTools/Weierstrass.md [hunk 3]
```diff
@@ -20,23 +22,81 @@ Then form a ``\mathbb{P}^{2,3,1}``-bundle over ``B3`` such that
...
-SpecificGlobalWeierstrassModel(f::MPolyElem_dec{fmpq, fmpq_mpoly}, g::MPolyElem_dec{fmpq, fmpq_mpoly}, base::Oscar.AbstractNormalToricVariety)
+GlobalWeierstrassModel(base::Oscar.AbstractNormalToricVariety)
+GlobalWeierstrassModelOverProjectiveSpace()
+GlobalWeierstrassModel(f::MPolyElem_dec{fmpq, fmpq_mpoly}, g::MPolyElem_dec{fmpq, fmpq_mpoly}, base::Oscar.AbstractNormalToricVariety)
+```
+
+## ... over not fully specified bases
+
+A global Weierstrass model can also be constructed over a base space that
+is not fully specified. Rather, it assums that a base space exists such that
```
Comment (apturner) [inline docs/src/FTheoryTools/Weierstrass.md:53]:
```suggestion
is not fully specified. Rather, it assumes that a base space exists such that
```

**Review by apturner — APPROVED** [review]
Included comments from #9 and #10 where they remained. Thanks Martin!

**Comment by HereAround** [comment]
@apturner  Indeed, the typo in the Tate polynomial was already being fixed in the code optimization of this PR.

**Pull request merged**